extensions/hibernate-search-orm-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/orm/elasticsearch/runtime/devconsole/HibernateSearchDevConsoleRecorder.java

```diff
@@ -0,0 +1,41 @@
+package io.quarkus.hibernate.search.orm.elasticsearch.runtime.devconsole;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.hibernate.search.mapper.orm.entity.SearchIndexedEntity;
+import org.hibernate.search.mapper.orm.mapping.SearchMapping;
+
+import io.quarkus.devconsole.runtime.spi.DevConsolePostHandler;
+import io.quarkus.runtime.annotations.Recorder;
+import io.vertx.core.Handler;
+import io.vertx.core.MultiMap;
+import io.vertx.ext.web.RoutingContext;
+
+@Recorder
+public class HibernateSearchDevConsoleRecorder {
+
+    public Handler<RoutingContext> indexEntity() {
+        return new DevConsolePostHandler() {
+            @Override
+            protected void handlePostAsync(RoutingContext event, MultiMap form) throws Exception {
+                SearchMapping searchMapping = HibernateSearchSupplier.searchMapping();
+                List<CompletableFuture<?>> indexationTasks = new ArrayList<>();
+                for (SearchIndexedEntity<?> indexedEntity : searchMapping.allIndexedEntities()) {
+                    if (form.contains(indexedEntity.jpaName())) {
+                        CompletionStage<?> indexationTask = searchMapping.scope(indexedEntity.javaClass()).massIndexer()
+                                .start();
+                        indexationTasks.add(indexationTask.toCompletableFuture());
+                    }
+                }
+                for (CompletableFuture<?> indexationTask : indexationTasks) {
+                    indexationTask.join();
+                }
```

Comment:
This whole looping will be sub-optimal. The mass indexer was intended to index multiple entities at once. Try building a scope that includes multiple types using `org.hibernate.search.mapper.orm.mapping.SearchMapping#scope(java.lang.Class<T>, java.util.Collection<java.lang.String>)`? You can pass `Object.class` as the type, as long as the strings are the entity names returned by `indexedEntity.jpaName()`.

E.g.:

```
searchMapping.scope(Object.class,
        searchMapping.allIndexedEntities().stream()
                .map(SearchIndexedEntity::jpaName)
                .filter(form::contains)
                .collect(Collectors.toList()))
        .massIndexer()
        .startAndWait();
```

Just so you know (definitely not the scope of this PR), there is also support for adding a monitor to the mass indexer, so in theory you could display the mass indexing progress on the webpage. I suppose you'd need some sort of push notifications for that, though. Or some global variable to store progress on the server-side and polling on the client side.

Reply:
Indeed this is way much cleaner. I pushed an update.

Yes, I suppose this could be done using SSE